output/fluentdforward/chunkencoder.go

```diff
@@ -26,7 +32,7 @@ func newEncoder(tag string, asArray bool, msgpackBufferSize int) *chunkEncoder {
 	}
 }
 
-func (enc *chunkEncoder) EncodeChunk(data []byte, params *shared.EncodeChunkParams) ([]byte, error) {
+func (enc *chunkEncoder) EncodeChunk(data []byte, params *encodeChunkParams) ([]byte, error) {
```

Comment:
Can use a TODO here for future optimization.

The way EncodeChunk works now is to encode last depending on the mode selected, but on production we always use the binary mode which is just a copying operation. It's a waste of resource and should be changed to be similar to datadog's, though fluentd output is no longer a priority.